Web Inspector: Incremental in FilterBar

francescorn opened the pull request:
#### 5df588e41bbde364c506774ce20012536c801b49
<pre>
Web Inspector: Incremental in FilterBar
<a href="https://bugs.webkit.org/show_bug.cgi?id=260043">https://bugs.webkit.org/show_bug.cgi?id=260043</a>

Reviewed by Devin Rousso.

The filter bar was not working correctly due to &quot;search&quot; being removed.

Replace incremental with throttler. Delete incremental getter and setter.

Switch eventlistener to &quot;input&quot; instead of &quot;search&quot;.

Rename &quot;_handleFilterSearched&quot; to &quot;_handleFilterInput&quot;

Delete unused &quot;_handleFilterInputEvent&quot;

* Source/WebInspectorUI/UserInterface/Views/FilterBar.js:
(WI.FilterBar.prototype.set filters):
(WI.FilterBar.prototype.addFilterBarButton):
(WI.FilterBar.prototype._handleFilterButtonToggled):
(WI.FilterBar.prototype._handleFilterInputEvent):
(WI.FilterBar.prototype.get incremental): Deleted.
(WI.FilterBar.prototype.set incremental): Deleted.
(WI.FilterBar.prototype._handleFilterChanged): Deleted.
(WI.FilterBar.prototype._handleFilterInputEvent): Deleted.

Canonical link: <a href="https://commits.webkit.org/266802@main">https://commits.webkit.org/266802@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0c8a8430b23c4f5a2ab6135e974ad983b8917525

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/14801 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/15106 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/15458 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/16548 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/13934 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/17627 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/15206 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/16577 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/14981 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/15475 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/12571 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/17282 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/12755 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/13354 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/20330 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/13833 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/13521 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/16757 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/14067 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/11887 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/13358 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3569 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/17691 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/13910 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/FilterBar.js:
```diff
@@ -174,7 +165,6 @@ WI.FilterBar = class FilterBar extends WI.Object
     }
 
     // Private
-
```
oops, please undo

francescorn replied:
Added the line back.

dcrousso replied:
it has extra whitespace now 😞

francescorn replied:
Fixed

dcrousso replied:
im still seeing it 😕

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/FilterBar.js:
```diff
@@ -38,9 +38,10 @@ WI.FilterBar = class FilterBar extends WI.Object
+        this._inputFieldInputThrottler = new Throttler(this._handleFilterInputEvent.bind(this), 250);
+        this._inputField.addEventListener("input", (event) => {
+            this._inputFieldInputThrottler.fire(event);
+        }, false);
```
i think you actually have this backwards.  the old `_handleFilterChanged` is what should now be used to handle `"input"` instead of `_handleFilterInputEvent` (and we can probably entirely get rid of `_handleFilterInputEvent`)

i'd remove `_handleFilterInputEvent` entirely and then rename `_handleFilterChanged` to `_handleFilterInput` (since it's now for an `"input"` instead of `"search"`)
```
this._handleFilterInputThrottler = new Throttler(this._handleFilterInput.bind(this), 250);
this._inputField.addEventListener("input", (event) => {
    if (this._inputField.value)
        this._handleFilterInputThrottler.fire(event);
    else
        this._handleFilterInputThrottler.force(event);
});
```
note that we do the `force` in order to mimic the old behavior where clearing the `WI.FilterBar` would immediately trigger instead of on the delay from `"search"`

nit: we dont need the `, false` in this case

francescorn replied:
Thank you for the explanation, added the code and renamed _handleFilterChanged to _handleFilterInput.

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/FilterBar.js:
```diff
@@ -202,13 +196,8 @@ WI.FilterBar = class FilterBar extends WI.Object
 
     _handleFilterInputEvent(event)
```
you can delete this entire function now since it's not used anymore

francescorn replied:
Removed _handleFilterInputEvent

dcrousso reviewed (approved):
nice!  😃

francescorn committed:
Web Inspector: Incremental in FilterBar
https://bugs.webkit.org/show_bug.cgi?id=260043

Reviewed by Devin Rousso.

The filter bar was not working correctly due to "search" being removed.

Replace incremental with throttler. Delete incremental getter and setter.

Switch eventlistener to "input" instead of "search".

Rename "_handleFilterSearched" to "_handleFilterInput"

Delete unused "_handleFilterInputEvent"

* Source/WebInspectorUI/UserInterface/Views/FilterBar.js:
(WI.FilterBar.prototype.set filters):
(WI.FilterBar.prototype.addFilterBarButton):
(WI.FilterBar.prototype._handleFilterButtonToggled):
(WI.FilterBar.prototype._handleFilterInputEvent):
(WI.FilterBar.prototype.get incremental): Deleted.
(WI.FilterBar.prototype.set incremental): Deleted.
(WI.FilterBar.prototype._handleFilterChanged): Deleted.
(WI.FilterBar.prototype._handleFilterInputEvent): Deleted.

Canonical link: https://commits.webkit.org/266802@main
